docs/content/src/learn/guides/saas-clients/sfdc46/salesforce-to-mysql-db/src/guide/main.bal

```diff
@@ -0,0 +1,150 @@
+import wso2/sfdc46;
+
+// Salesforce configuration.
+sfdc46:SalesforceConfiguration sfConfig = {
+    baseUrl: config:getAsString("SF_BASE_URL"),
+    clientConfig: {
+        accessToken: config:getAsString("SF_ACCESS_TOKEN"),
+        refreshConfig: {
+            clientId: config:getAsString("SF_CLIENT_ID"),
+            clientSecret: config:getAsString("SF_CLIENT_SECRET"),
+            refreshToken: config:getAsString("SF_REFRESH_TOKEN"),
+            refreshUrl: config:getAsString("SF_REFRESH_URL")
+        }
+    }
+};
+
+// MySQL configuration.
+jdbc:Client contactsDB = new ({
+    url: config:getAsString("JDBC_URL"),
+    username: config:getAsString("DB_USERNAME"),
+    password: config:getAsString("DB_PASSWORD"),
+    poolOptions: {maximumPoolSize: 5},
+    dbOptions: {useSSL: false}
+});
+
+// Represents `contacts` table.
+type Contact record {
+    string email;
+    string first_name;
+    string last_name;
+};
+
+// Create salesforce client.
+sfdc46:Client salesforceClient = new (sfConfig);
+
+public function main() returns error? {
+    task:Scheduler timer = new ({
+        intervalInMillis: config:getAsInt("SCHEDULER_INTERVAL_IN_MILLIS"),
+        initialDelayInMillis: 0
+    });
+
+    // Attach the service to the scheduler.
+    var attachResult = timer.attach(sfdcToMysqlService);
+    if (attachResult is error) {
+        log:printError("Error attaching the sfdcToMysqlService.", attachResult);
+        return;
+    }
+
+    // Start the scheduler.
+    var startResult = timer.start();
+    if (startResult is error) {
+        log:printError("Starting the task is failed.", startResult);
+        return;
+    }
+}
+
+service sfdcToMysqlService = service {
+    resource function onTrigger() returns error? {
+        log:printInfo("service started...");
+        sfdc46:SoqlRecord[] newSfContacts = check getNewSfContacts();
+        if (updateDb(newSfContacts)) {
+            log:printInfo("Batch job SFDC -> MySQL has been completed.");
+        } else {
+            log:printError("Batch job SFDC -> MySQL has been failed!");
+        }
+    }
+};
+
+function getNewSfContacts() returns @tainted sfdc46:SoqlRecord[]|error {
+    string q = "SELECT Email, FirstName, LastName, LastModifiedDate FROM Contact WHERE LastModifiedDate > YESTERDAY";
+    sfdc46:SoqlResult queryResult = check salesforceClient->getQueryResult(q);
+    
+    if (queryResult.done == true) {
+        return queryResult.records;
+    } else {
+        return error("Query failed!");
+    }
+}
+
+function updateDb(sfdc46:SoqlRecord[] newSfContacts) returns boolean {
+    foreach sfdc46:SoqlRecord newSfContact in newSfContacts {
+        // Query DB and select contacts using the `email`.
+        string q = "SELECT first_name, last_name, email FROM contacts WHERE email=?";
+        table<Contact> | error dbRes = contactsDB->select(<@untainted> q, Contact, newSfContact["Email"].toString());
+
+        if (dbRes is table<Contact>) {
+            if (dbRes.hasNext()) {
```

Comment:
We are not iterating the table data fully here. So this table won't get closed (i.e. there will be an underlying open DB connection ). To avoid that we need to explicitly close the table using `dbRes.close()` to avoid this connection leak